runtime/src/main/java/uk/ac/manchester/tornado/runtime/graal/phases/TornadoReduceReplacement.java

```diff
@@ -149,28 +149,23 @@ private boolean checkIfVarIsInLoop(StoreIndexedNode store) {
 
     }
 
-    private ValueNode obtainInputArray(ValueNode currentNode, ValueNode outputArray, ValueNode indexToStore) {
+    private ValueNode obtainInputArray(ValueNode currentNode, ValueNode outputArray) {
         ValueNode array = null;
         if (currentNode instanceof StoreIndexedNode) {
             StoreIndexedNode store = (StoreIndexedNode) currentNode;
-            return obtainInputArray(store.value(), store.array(), store.index());
+            return obtainInputArray(store.value(), store.array());
         } else if (currentNode instanceof BinaryArithmeticNode) {
             BinaryArithmeticNode<?> value = (BinaryArithmeticNode<?>) currentNode;
-            array = obtainInputArray(value.getX(), outputArray, indexToStore);
+            array = obtainInputArray(value.getX(), outputArray);
             if (array == null) {
-                array = obtainInputArray(value.getY(), outputArray, indexToStore);
+                array = obtainInputArray(value.getY(), outputArray);
             }
         } else if (currentNode instanceof BinaryNode) {
             String currentNodeName = currentNode.getClass().getName();
             if (currentNodeName.endsWith(OCL_FP_BINARY_NODE) || currentNodeName.endsWith(PTX_FP_BINARY_NODE)) {
-                array = obtainInputArray(((BinaryNode) currentNode).getX(), outputArray, indexToStore);
+                array = obtainInputArray(((BinaryNode) currentNode).getX(), outputArray);
                 if (array == null) {
-                    array = obtainInputArray(((BinaryNode) currentNode).getY(), outputArray, indexToStore);
-                }
-            } else if (currentNodeName.endsWith(OCL_FP_BINARY_NODE) || currentNodeName.endsWith(PTX_FP_BINARY_NODE)) {
```

Comment:
the `OCL_FP_BINARY_NODE` accidentally replaced the `OCLIntBinaryIntrinsicNode` in a previous merge.
I guess the else if should have `OCL_INT_BINARY_NODE` instead of `OCL_FP_BINARY_NODE`. 

Can we check if this `else if` condition is needed? Otherwise we can delete it (as it is).

Reply:
Good catch. I refactor this part.